src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsEmulatorConfigWriterHook.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Lifecycle;
+using System.Text.Json;
+
+namespace Aspire.Hosting.Azure;
+
+internal sealed class AzureEventHubsEmulatorConfigWriterHook : IDistributedApplicationLifecycleHook
+{
+    public Task AfterEndpointsAllocatedAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken)
+    {
+        if (appModel.Resources.OfType<AzureEventHubsResource>().SingleOrDefault() is not { } emulatorResource || !emulatorResource.IsEmulator)
+        {
+            // No-op if there is no Azure Event Hubs emulator resource.
+            return Task.CompletedTask;
+        }
+
+        var configFileMount = emulatorResource.Annotations.OfType<ContainerMountAnnotation>().Single(v => v.Target == AzureEventHubsEmulatorResource.EmulatorConfigJsonPath);
+
+        if (!emulatorResource.Hubs.Any())
+        {
+            // No-op if there are no Azure Event Hubs resources present.
+            return Task.CompletedTask;
+        }
+
+        using var stream = new FileStream(configFileMount.Source!, FileMode.Create);
+        using var writer = new Utf8JsonWriter(stream);
+
+        writer.WriteStartObject();                      // {
+        writer.WriteStartObject("UserConfig");          //   "UserConfig": {
+        writer.WriteStartArray("NamespaceConfig");      //     "NamespaceConfig": [
+        writer.WriteStartObject();                      //       {
+        writer.WriteString("Type", "EventHub");         //         "Type": "EventHub",
+        writer.WriteString("Name", "emulatorNs1");      //         "Name": "emulatorNs1"
+        writer.WriteStartArray("Entities");             //         "Entities": [
+
+        foreach (var hub in emulatorResource.Hubs)
+        {
+            // The default consumer group ('$default') is automatically created, and this
+            // is also the one used by a consumer client if no ConsumerGroup value
+            // is provided.
+            writer.WriteStartObject();                  //           {
+            writer.WriteString("Name", hub.Name);       //             "Name": "hub",
+            writer.WriteString("PartitionCount", "2");  //             "PartitionCount": "2",
+            writer.WriteStartArray("ConsumerGroups");   //              "ConsumerGroups": [
+            writer.WriteStartObject();                  //                {
+            writer.WriteString("Name", "cg1");          //                  "Name": "cg1"
```

Comment:
What does hard-coding `cg1` here mean? Do you have to have a consumer group?

Reply:
There was a bug I reported that if would throw if the array was empty. I just want the default here. I will update since it just got fixed and published.

Reply:
Done